inc/ThirdParty/Plugins/Ecommerce/XLWooCommerceSalesTriggersSubscriber.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace WP_Rocket\ThirdParty\Plugins\Ecommerce;
+
+use WP_Rocket\Engine\Optimization\DelayJS\HTML;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+class XLWooCommerceSalesTriggersSubscriber implements Subscriber_Interface {
+
+	/**
+	 * Delay JS HTML class.
+	 *
+	 * @var HTML
+	 */
+	private $delayjs_html;
+
+	/**
+	 * WooCommerceSubscriber constructor.
+	 *
+	 * @param HTML $delayjs_html DelayJS HTML class.
+	 */
+	public function __construct( HTML $delayjs_html ) {
+		$this->delayjs_html = $delayjs_html;
+	}
+
+	/**
+	 * Get the subscribed events array.
+	 *
+	 * @return array|array[]
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'rocket_delay_js_exclusions' => [ 'modify_delayjs_exclusions_on_product_pages', 10, 2 ],
+		];
+	}
+
+	/**
+	 * Modify delay JS excluded scripts for compatibility.
+	 *
+	 * @param array $exclusions The initial Delay JS exclusion list.
+	 *
+	 * @return array The filtered Delay JS exclusion list.
+	 */
+	public function modify_delayjs_exclusions_on_product_pages( array $exclusions ) {
+		if ( empty( rocket_get_constant( 'WCST_VERSION' ) ) ) {
+			return $exclusions;
+		}
+
+		if ( ! is_product() || ! is_single() ) {
```

Comment:
won't the `is_single()` exclude jquery on any post type single page?

Reply:
The logic here is, we're going to bail out whenever it's NOT a product OR whenever it's NOT single. So, I think, checking against not/OR in both sides will allow it to add the exclusions only when it IS a product AND IS single.

Reply:
Checking back, `is_single()` only applies to the `post` post type, not all of them (`is_singular()` does).

Also, the underlying code for `is_product()` is `is_singular( array( 'product' ) )` so in the end the "single" check is already done at that point.

So I would suggest only keeping `is_product()` there, as it's going to check everything needed, what do you think?

Reply:
Indeed, that will work. Thanks.